reports/tests.py

```diff
@@ -0,0 +1,504 @@
+            self.assertIsInstance(instance, OpenEdXInstance)
+
+    @mock.patch('reports.helpers.get_instance_charges')
+    def test_generate_charges(self, instance_charges_mock):
+        """
+        This method is to make sure the we're able to generate the charges
+        for all appservers in all forks of an organization.
+        :param instance_charges_mock: A mock for get_instance_charges function.
+        """
+        invoice_month = self._generate_invoice_month(2017)
+        instance_charges = 22
+        charges_mock = {
+            'name': 'Mock AppServer',
+            'billing_start': invoice_month,
+            'billing_end': invoice_month,
+            'days': 10,
+            'charge': 31,
+        }
+        instance_charges_mock.return_value = ([charges_mock], instance_charges)
+        watched_forks = {
+            'fork1': [OpenEdXInstanceFactory() for _ in range(2)],
+            'fork2': [OpenEdXInstanceFactory() for _ in range(1)],
+        }
+
+        appservers_charges, appservers_total = generate_charges(watched_forks, invoice_month)
+        self.assertEqual(appservers_total, 3 * instance_charges)
+        self.assertEqual(len(appservers_charges), 2)
+        self.assertEqual(len(appservers_charges['fork1']['instances']), 2)
+        self.assertEqual(len(appservers_charges['fork2']['instances']), 1)
+        self.assertIn(str(charges_mock), str(appservers_charges['fork1']['instances']))
+        self.assertIn(str(charges_mock), str(appservers_charges['fork2']['instances']))
+
+    def _setup_watched_forks(self):
+        """
+        This is a helper method that will generate the mesh of forks,
+        instances, and pull requests linked together
+        """
+        watched_fork1 = WatchedForkFactory.create(
+            fork='fork/repo1',
+            organization=self.organization)
+        watched_fork2 = WatchedForkFactory.create(
+            fork='fork/repo2',
+            organization=self.organization)
+
+        instance1 = OpenEdXInstanceFactory()
+        instance2 = OpenEdXInstanceFactory()
+        instance3 = OpenEdXInstanceFactory()
+
+        WatchedPullRequest.objects.create(
+            fork_name=watched_fork1.fork,
+            branch_name='new-tag',
+            ref_type='tag',
+            watched_fork=watched_fork1,
+            instance=instance1
+        )
+        WatchedPullRequest.objects.create(
+            fork_name=watched_fork1.fork,
+            branch_name='new-tag',
+            ref_type='tag',
+            watched_fork=watched_fork1,
+            instance=instance2
+        )
+        WatchedPullRequest.objects.create(
+            fork_name=watched_fork2.fork,
+            branch_name='new-tag',
+            ref_type='tag',
+            watched_fork=watched_fork2,
+            instance=instance3
+        )
+
+        return watched_fork1, watched_fork2
+
+    @staticmethod
+    def _generate_invoice_month(year=2018, month=1, this_month=False):
```

Comment:
I updated the documentation as suggested, but kept it generate because I don't want it to imply that we're getting a date that already exists in somewhere else.